### PR TITLE
update subtask list

### LINKED ISSUE #391: SubtaskList
### Describe the feature

SubtaskList is ListView for Subtasks, can accept a list of subtasks or a taskId, first uses the subtasks while the latter loads them by the taskId.
Update are notified by an onChange, and when the taskId is given are directly (or with a 3 sec delay) propagated to the backend.

### Why is this feature needed?

To have a seperate component for this.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #391

### CODE REVIEW

**Comment by DasProffi** [comment]
Waiting for #432

**Comment by MaxSchaefer** [comment]
> Waiting for #432

@DasProffi #432 is closed.

**Comment by DasProffi** [comment]
Changing the name of Subtasks will be done seperately:
See #448

File: apps/tasks/lib/controllers/subtask_list_controller.dart [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  }
+
+  /// Add the [SubTask]
+  Future<void> add(SubTask subTask) async {
+    if (isCreating) {
+      _subtasks.add(subTask);
+      notifyListeners();
+      return;
+    }
+    TaskService().addSubTask(taskId: taskId, subTask: subTask).then((value) {
```
Comment (MaxSchaefer) [inline apps/tasks/lib/controllers/subtask_list_controller.dart:106]:
```suggestion
    await TaskService().addSubTask(taskId: taskId, subTask: subTask).then((value) {
```

File: apps/tasks/lib/controllers/subtask_list_controller.dart [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      return null;
+    });
+  }
+
+  Future<void> changeStatus({required SubTask subTask, required bool value}) async {
+    if (subTask.isCreating) {
+      subTask.isDone = value;
+    } else {
+      state = LoadingState.loading;
+      TaskService().changeSubtaskStatus(id: subTask.id, isDone: value).then((value) {
```
Comment (MaxSchaefer) [inline apps/tasks/lib/controllers/subtask_list_controller.dart:121]:
```suggestion
      await TaskService().changeSubtaskStatus(id: subTask.id, isDone: value).then((value) {
```

File: apps/tasks/lib/controllers/subtask_list_controller.dart [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        return null;
+      });
+    }
+    notifyListeners();
+  }
+
+  Future<void> updateSubtask({required SubTask subTask}) async {
+    if (!subTask.isCreating) {
+      state = LoadingState.loading;
+      TaskService().updateSubTask(subTask: subTask).then((value) {
```
Comment (MaxSchaefer) [inline apps/tasks/lib/controllers/subtask_list_controller.dart:136]:
```suggestion
      await TaskService().updateSubTask(subTask: subTask).then((value) {
```

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**